src/type-data-frame.c

```diff
@@ -0,0 +1,57 @@
+#include "vctrs.h"
+#include "utils.h"
+
+
+// [[ include("utils.h") ]]
+SEXP new_data_frame(SEXP x, R_len_t n) {
+  x = PROTECT(r_maybe_duplicate(x));
+
+  init_data_frame(x, n);
+
+  if (Rf_length(x) == 0) {
+    Rf_setAttrib(x, R_NamesSymbol, vctrs_shared_empty_chr);
```

Comment:
Why isn't this done in `init_data_frame()`?

Reply:
Good point.